### ISSUE TITLE
Make DeveloperExceptionPage log the actual exception, so it can also be viewed in the local console

### ISSUE THREAD

**Issue by zlepper** [dotnet/aspnetcore#33026]
### Is your feature request related to a problem? Please describe.
Background: Locally we develop several micro-services at the same time. When we do that, they are running in full development mode, including the DeveloperExceptionPage. This page is really awesome when we are calling a given service directly and can just view the exceptions in our browser. However because we have microservices comunicating internally, the actual error page is often lost somewhere between all the network calls. 

The actual problem we encounter is that, even though DeveloperExceptionPage logs an error and returns the full error in the network response, it doesn't log the actual underlying error, which means we have to either disable the middleware to get the actual error or intercept and dump the network response body in the client service, neither of which are particularly nice. 

I do have a feeling i can hook into the `Microsoft.AspNetCore.Diagnostics.UnhandledException` diagnostic event, and log the exception there, however it would be considerably easier if the middleware by itself just logged in actual exception so it could see it in the console.

### Describe the solution you'd like
The `DeveloperExceptionPageMiddleware` should log the actual exception. 

I do have a feeling i can hook into the `Microsoft.AspNetCore.Diagnostics.UnhandledException` diagnostic event, and log the exception there, however it would be considerably easier if the middleware by itself just logged in actual exception so it could see it in the console.

We could also create our own middleware that runs just before `DeveloperExceptionPage` and logs the error, but again it feels kinda dump when AspNetCore does log the error by default in production.

**Comment by Tratcher** [comment]
It's logged here:
https://github.com/dotnet/aspnetcore/blob/e7b5aa6f713e9f040ba0730b915ae407d35971c1/src/Middleware/Diagnostics/src/DeveloperExceptionPage/DeveloperExceptionPageMiddleware.cs#L98
https://github.com/dotnet/aspnetcore/blob/e7b5aa6f713e9f040ba0730b915ae407d35971c1/src/Middleware/Diagnostics/src/DiagnosticsLoggerExtensions.cs#L14

What do you see in the logs?

**Comment by zlepper** [comment]
I only see `An unhandled exception has occurred while executing the request`. 

I did also arrive at the conclusion, that it was passed somehow, but wasn't actually outputted. I think it's because the format operation doesn't actual access the exception to output it, which normally happens for things like entity framework. This compounded by the facts that we have a custom logger (Due to various missing features of the normal microsoft logger), and expected that since we have to pass both `state` and `exception` to the passed in formatter, that the function would then add the exception there also. Apparently not the case for this particular instance. 

I have updated out own logger to explicitly output the exception if one is passed in, ignoring what the formatter does.
